scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,227 @@
+package org.oppia.android.scripts
+
+import com.google.protobuf.MessageLite
+import org.oppia.android.scripts.proto.FileContentCheck
+import org.oppia.android.scripts.proto.FileContentChecks
+import org.oppia.android.scripts.proto.FilenameCheck
+import org.oppia.android.scripts.proto.FilenameChecks
+import java.io.File
+import java.io.FileInputStream
+
+/**
+ * Script for ensuring that prohibited file contents and
+ * file naming patterns are not present in the codebase.
+ */
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+      // path of the repo to be analyzed.
+      val repoPath = args[0] + "/"
+
+      // a list of all allowed directories in the repo to be analyzed.
+      // args[0] is the repoPath, the allowed directories are specified
+      // after it. Hence, we have to start from the 1st index.
+      val allowedDirectories = args.drop(1)
+
+      // a list of all files in the repo to be analyzed.
+      val searchFiles = RepoFile.collectSearchFiles(
+        repoPath,
+        allowedDirectories
+      )
+
+      // class object which is needed to acess the helper methods
+      val regexPatternValidationCheck: RegexPatternValidationCheck =
+        RegexPatternValidationCheck()
```

Comment:
```suggestion
      val regexPatternValidationCheck = RegexPatternValidationCheck()
```

Can be simplified a bit.

Reply:
Done.